test/e2e/node/pods.go

```diff
@@ -275,7 +275,9 @@ var _ = SIGDescribe("Pods Extended", func() {
 						ch := make(chan []watch.Event)
 						go func() {
 							defer close(ch)
-							w, err := podClient.Watch(context.TODO(), metav1.ListOptions{
+							ctx, cancel := context.WithTimeout(context.Background(), framework.SingleCallTimeout)
```

Comment:
@liggitt I think that the problem with this test is that it may get stuck in L298
```go
ch <- events
```

if the context cancels the Watch by timeout, it will close the channel and end the goroutine?

Reply:
there's a `defer close(ch)` when that goroutine returns

Reply:
and the thing receiving from the channel handles a closed channel:
```
events, ok := <-ch
if !ok {
   continue
}
```

Reply:
I'm more concerned about a scenario where the watch encounters and error and does not close